services/core-api/app/api/mines/tailings/resources/tailings.py

```diff
@@ -48,67 +37,39 @@ class MineTailingsStorageFacilityListResource(Resource, UserMixin):
         help='Operating Status of the storage facility',
         required=True)
     parser.add_argument(
-        'has_itrb', type=bool, trim=True, help='Risk Severity Classification', required=True)
+        'has_itrb',
+        type=lambda x: True if x.lower() in ("true") else False,
+        trim=True,
+        help='Risk Severity Classification',
+        required=True)
     parser.add_argument(
         'eor_party_guid',
         type=str,
         help='GUID of the party that is the Engineer of Record for this TSF.',
         location='json',
         store_missing=False)
 
-    @api.doc(description='Gets the tailing storage facilites for the given mine')
-    @api.marshal_with(
-        MINE_TSF_MODEL, envelope='mine_tailings_storage_facilities', as_list=True, code=200)
-    @requires_role_view_all
-    def get(self, mine_guid):
-        mine = Mine.find_by_mine_guid(mine_guid)
-        if not mine:
-            raise NotFound('Mine not found.')
-        return mine.mine_tailings_storage_facilities
-
-    @api.doc(description='Creates a new tailing storage facility for the given mine')
-    @api.marshal_with(MINE_TSF_MODEL, code=201)
+    @api.doc(description='Updates an existing tailing storage facility for the given mine')
     @requires_role_mine_edit
-    def post(self, mine_guid):
+    @api.marshal_with(MINE_TSF_MODEL)
+    def put(self, mine_guid, mine_tailings_storage_facility_guid):
         mine = Mine.find_by_mine_guid(mine_guid)
+
         if not mine:
             raise NotFound('Mine not found.')
 
-        data = self.parser.parse_args()
-
-        mine_tsf_list = mine.mine_tailings_storage_facilities
-        is_mine_first_tsf = len(mine_tsf_list) == 0
+        mine_tsf = MineTailingsStorageFacility.find_by_tsf_guid(mine_tailings_storage_facility_guid)
 
-        mine_tsf = MineTailingsStorageFacility.create(
-            mine,
-            mine_tailings_storage_facility_name=data['mine_tailings_storage_facility_name'],
-            latitude=data['latitude'],
-            longitude=data['longitude'],
-            consequence_classification_status_code=data['consequence_classification_status_code'],
-            has_itrb=data['has_itrb'],
-            tsf_operating_status_code=data['tsf_operating_status_code'])
-        mine.mine_tailings_storage_facilities.append(mine_tsf)
+        if not mine_tsf:
+            raise NotFound("Tailing Storage Facility not found")
 
-        if is_mine_first_tsf:
```

Comment:
I'm guessing it was okayed to get rid of this logic?

Reply:
This is fixed.